### DISCUSSION TITLE
BTC migrating nbxplorer from dbtrie to postgresql...

### DISCUSSION THREAD

**Discussion by SireWilliamson** [btcpayserver/btcpayserver#4931] [Technical Support]
I have the commands and variables I need to migrate, but I'm not sure of the actual process. If I try to migrate, when I'm running on postgresql, my wallet balance is zero.

Here is what I think the process is, please correct me where I am going wrong.
1. Make sure postgresql is installed (Mine is, I installed it for btcpay)
2. create a nbxplorer DB.
```
CREATE DATABASE nbxplorer TEMPLATE 'template0' LC_CTYPE 'C' LC_COLLATE 'C' ENCODING 'UTF8';
CREATE USER nbxplorer WITH ENCRYPTED PASSWORD 'urpassword';
GRANT ALL PRIVILEGES ON DATABASE nbxplorer TO nbxplorer;

```
3. Shutdown BTCPay
4. Shutdown NBxplorer
5. add connection string to nbxplorer.config
6. (This is where my first confusion starts.) I need to have the same connection string for NBXplorer and BTCPay, but my existing BTCPay connection string is different. So do I add both to my btcpay.config or do I replace my old connection string in btcpay.config with the one I just made?
7.  Start up nbxplorer.service with "--automigrate --postgresql --conf=..."
8. start btcpay
9. verify migration
10. stop btcpay
11. stop nbxplorer
12. nbxplorer/run.sh --deleteaftermigration
13. stop nbxplorer
14. start nbxplorer "--postgresql --conf=..."
15. start btcpay.
16.  Done

**Comment by pavlenex** [comment]
@NicolasDorier maybe you know where to point this to?
